### PR TITLE
Make it build with ghc-9.6

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: src/Validation.hs [hunk 1]
```diff
@@ -858,7 +858,7 @@ instance (NoValidationMonadError, Semigroup e) => Monad (Validation e) where
 type family NoValidationMonadError :: Constraint where
     NoValidationMonadError = TypeError
         ( 'Text "Type 'Validation' doesn't have lawful 'Monad' instance"
-        ':$$: 'Text "which means that you can't use 'Monad' methods with 'Validation'."
+        ' :$$: 'Text "which means that you can't use 'Monad' methods with 'Validation'."
```
Comment (erikd) [inline src/Validation.hs:861]: This was required because the parser for `ghc-9.6` **insists** on whitespace around operators.

**Comment by erikd** [comment]
Ping?

**Comment by Kleidukos** [comment]
@erikd Hi, and thank you for the PR. :) We've recently merged #82, would you mind rebasing on HEAD? :)

**Comment by erikd** [comment]
Rebased against master and made it build with `ghc-9.8` as well. 

Currently the tests fail with both versions of the compiler.

**Comment by tchoutri** [comment]
@erikd the default branch is `main` :)

**Comment by tomjaguarpaw** [comment]
Both of these work, so closing this as already completed. Thanks @erikd.

```
cabal repl -z -b validation-selective -w ghc-9.6
cabal repl -z -b validation-selective -w ghc-9.8
```

**Pull request closed without merging**